src/eventDispatcher.js

```diff
@@ -32,11 +35,15 @@ class EventDispatcher {
      *  @param {String} eventName. 
      */
     trigger(eventName) {
-        if (this.eventDispatcherMap[eventName]) {
-            for (let i = 0; i < this.eventDispatcherMap[eventName].length; i++) {
-                try {
-                    this.eventDispatcherMap[eventName][i]();
-                } catch (e) {}
+        if (typeof eventName !== 'string') {
+            throw new SyntaxError("Parameter is not string");
```

Comment:
Why are you using SyntaxError? There are many types of errors in ES6, check the list here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error. From what I see, the correct error for this case is `TypeError`.